### PR TITLE
spotifywm: behave as patch with same binary name as spotify

### PR DESCRIPTION
**Pull request by the-argus**
###### Description of changes

This change serves to make spotifywm behave as a wrapper/patch instead of being its own separate package using ``bin/spotifywm`` instead of ``bin/spotify``.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by spikespaz** [comment]
That looks good to me. If you can, also bump the source to use latest commit and update hashes.

**Review by jqueiroz — APPROVED** [review]

**Comment by the-argus** [comment]
> That looks good to me. If you can, also bump the source to use latest commit and update hashes.

There haven't been any commits to spotifywm since this package was made.

**Review by lilyinstarlight — COMMENTED** [review]
Looks good to me! Do you think you'd be able to go ahead and add `runHook preInstall` and `runHook postInstall` at the start and end of the `installPhase`, since generally those are supposed to be added when phases are overriden?

(Also I don't quite understand the reason there's a `propagatedBuildInputs` in this derivation? But that's out-of-scope here)

File: pkgs/applications/audio/spotifywm/default.nix [hunk 1]
```diff
@@ -10,22 +10,27 @@ stdenv.mkDerivation {
     sha256 = "sha256-AsXqcoqUXUFxTG+G+31lm45gjP6qGohEnUSUtKypew0=";
   };
 
-  buildInputs = [ xorg.libX11 ];
+  buildInputs = [ xorg.libX11 makeWrapper ];
```
Comment (Artturin) [inline pkgs/applications/audio/spotifywm/default.nix:13]: makeWrapper should be in nativeBuildInputs

**Comment by the-argus** [comment]
@lilyinstarlight @Artturin I've made the requested changes- see what you think.

**Review by Artturin — APPROVED** [review]

**Pull request merged**